### PR TITLE
Thermal RC network

### LINKED ISSUE #11: Model -> RC network conversion

### PR DESCRIPTION
**Pull request by bluecube**
TODO:
- [x] Figure out and implement convection effects on boundary surfaces
- [x] Manual smoke test
    - [x] Visualization / export to graphviz / something else to help debugging
- [x] Unit tests
- [x] Figure out a way to reference nodes from the network to allow external heat input ( = solar, room heating, ...)
- [x] Make "air" built-in (moved from #19 )

After #19 gets merged
- [x] Rebase on main
- [x] Processing marker boundary layers, hash map for accessing marked nodes between-layers.
- [x] Tests
- [x] <strike>Modelling ground -- There's no air in there!</strike> => #20 

Resolves #11

### CODE REVIEW

**Comment by bluecube** [comment]
When done, this will resolve #11

File: MPCHomeControl/src/rc_network.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+/// Add nodes corresponding to the boundary layers to the graph, including connections.
+fn add_layered_boundary_nodes(
+    graph: &mut UnGraph<Node, Edge>,
+    z1: NodeIndex,
+    z2: NodeIndex,
+    layers: &Vec<BoundaryLayer>,
+    area: Area,
+    convection_conductance: ThermalConductance,
+) {
+    let mut nodes = Vec::with_capacity(layers.len() + 1);
```
Comment (bluecube) [inline MPCHomeControl/src/rc_network.rs]: Vidis jak jsem se udrzel a nenapsal to cele jednim velkym iteratorem? :-D

File: MPCHomeControl/src/model.rs [hunk 2]
```diff
@@ -149,6 +191,18 @@ pub struct BoundaryLayer {
     pub thickness: Length,
 }
 
+impl BoundaryLayer {
+    pub fn heat_capacity(&self, area: Area) -> HeatCapacity {
```
Comment (bluecube) [inline MPCHomeControl/src/model.rs]: Trochu se mi tady nelibi to ze musim predavat Area. Mas nejaky lepsi napad jak to udelat? Vracet HeatCapacity per square meter? nejak posunout tyto metody do `Boundary`?

File: MPCHomeControl/src/model.rs [hunk 3]
```diff
@@ -130,6 +166,12 @@ pub struct Boundary {
     pub area: Area,
 }
 
+impl Boundary {
+    pub fn convection_conductance(&self) -> ThermalConductance {
+        todo!("Figure out how to model this!")
```
Comment (bluecube) [inline MPCHomeControl/src/model.rs]: Hlavni problem teto verze. Vubec netusim jak se to pocita, jeste stale jsem nevidel ty videa .... :-( Pokud mas nejaky insight, tak sem s nim. Taky si nejsem jisty jestli (kdyz uz tuto hodnotu spocitam) ji pak spravne pouziju.

Take by se tu mozna mel nejak zohlednovat vitr (pripsal jsem to do #6 )

**Comment by bluecube** [comment]
graph


(vypada to trochu 💩, protoze ten model.json5 je docela rozbity)

**Comment by bluecube** [comment]
Ceka na merge #19 , pak se musi udelat velky a konfliktni rebase.
Neni to moc otestovane, nenapada me jak testovat: 1. konektivitu grafu, 2. Hodnoty (mam ale dobry pocit z toho ze UOM by me nenechalo udelat tam moc chyb), 3. Vystup do DOT.

**Review by tommarek — APPROVED** [review]

**Pull request merged**